File: 234-palindrome-linked-list/234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        List<Integer> list = new ArrayList<>();
        
        ListNode curr = head;
        
        while(curr != null) {
            list.add(curr.val);
            curr = curr.next;
        }
        
        int start = 0;
        int end = list.size() - 1;
        
        while(start < end) {
            if(list.get(start++) != list.get(end--)) {
                return false;
            }
        }
        
        return true;
    }
}

File: 234-palindrome-linked-list/234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        list = []
        curr = head
        while curr is not None:
            list.append(curr.val)
            curr = curr.next
        return list == list[::-1]